quota/mysql/mysql_quota.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mysql defines a MySQL-based quota.Manager implementation.
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/google/trillian/quota"
+)
+
+const (
+	// DefaultMaxUnsequenced is a suggested value for MaxUnsequencedRows.
+	// Note that this is a Global/Write quota suggestion, so it applies across trees.
+	DefaultMaxUnsequenced = 500000 // About 2h of non-stop signing at 70QPS.
+)
+
+// QuotaManager is a MySQL-based quota.Manager implementation.
+//
+// It has two working modes: one queries the information schema for the number of Unsequenced rows,
+// the other does a select count(*) on the Unsequenced table. Information schema queries are
+// default, even though they are approximate, as they're constant time (select count(*) on InnoDB
+// based MySQL needs to traverse the index and may take quite a while to complete).
```

Comment:
Yep, I've considered this option too, but I would prefer to avoid a single point of contact if possible.

The `count(*)` mode is present mostly so we have fully deterministic tests. As I stated in a comment above, all tests *seem* to run fine on the information_schema mode, but I'm afraid they might get flaky in other environments (other people's machines, Travis, so on). I could completely remove the `count(*)` mode, if you prefer, in which case we would deal with flakiness if/when it happens.